Tools/gsc/finalize_manifests.py

```diff
@@ -0,0 +1,154 @@
+                num_trusted += 1
+
+    print('Found ' + str(num_trusted) + ' files in \'' + root_dir + '\'.')
+
+    return trusted_files
+
+def generate_library_paths():
+    ld_paths = subprocess.check_output('ldconfig -v',
+           stderr=subprocess.PIPE, shell=True).decode().splitlines()
+
+    # Library paths start without whitespace. Libraries found under a path start with an
+    # indentation.
+    ld_paths = (line for line in ld_paths if not re.match(r'(^\s)', line))
+
+    return ''.join(ld_paths)
+
+def get_binary_path(executable):
+    path = subprocess.check_output('which ' + executable,
+           stderr=subprocess.STDOUT, shell=True).decode()
+    return path.replace('\n', '')
+
+def generate_signature(manifest):
+    sign_process = subprocess.Popen([
+        '/graphene/signer/pal-sgx-sign',
+        '-libpal', '/graphene/Runtime/libpal-Linux-SGX.so',
+        '-key', '/graphene/signer/enclave-key.pem',
+        '-output', manifest + '.sgx',
+        '-manifest', manifest
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+
+    _, err = sign_process.communicate()
+
+    if (sign_process.returncode != 0
+        or not os.path.exists(os.path.join(os.getcwd(), manifest + '.sgx'))):
+        print(err.decode())
+        print("Finalize manifests failed due to pal-sgx-sign failure.")
+        sys.exit(1)
+
+# Iterate over manifest file to find enclave size definition and return it
+def extract_enclave_size(manifest):
+    with open(manifest, "r") as file:
+        for line in file:
+            if not line.strip().startswith("sgx.enclave_size"):
+                continue
+
+            tokens = line.split("=")
+            if len(tokens) != 2 or "#" in tokens[1]:
+                continue
+            return tokens[1].strip()
+
+    return "0M"
+
+ARGPARSER = argparse.ArgumentParser()
+ARGPARSER.add_argument('finalize_manifests.py', metavar='SCRIPT',
+    help='Script to be run.')
+ARGPARSER.add_argument('directory', metavar='DIRNAME',
+    help='Search the directory tree from this root for files and generate list of trusted files')
+ARGPARSER.add_argument('manifests', metavar='APP.manifest',
+    nargs='+',
+    help='Application-specific manifest files. The first manifest will be used for the entry point '
+        'of the docker image. If file does not exist, manifest will be generated without '
+        'application-specific values.')
+
+def main(args=None):
+    args = ARGPARSER.parse_args(args)
+
+    if not os.path.isdir(args.directory):
+        ARGPARSER.error('Could not find directory ' + args.directory + '.')
+
+    trusted_files = generate_trusted_files(args.directory)
+    library_paths = generate_library_paths()
+
+    print('Setting LD_LIBRARY_PATH to \'' + library_paths + '\'.')
+
+    trusted_signatures = []
+
+    # To deal with multi-process applications, we allow multiple manifest files to be specified.
```

Comment:
@vahldiek Again we discussed this topic today, but adding comment for reference. Currently GSC tool sets up parent-child process relationship through trusted_child parameters, using over-estimation design. So current design, can introduce new parent-child trust relationships, not intended by the original unmodified application's design. Given that this has security risk, this is something that needs to be documented and fixed in a later version, if not now.